File: Chapter 03/JS/3.3.5 DIP/Before/coach.js
const Forward = require('./forward');
const Midfielder = require('./midfielder');

class Coach {
    constructor() {
        this.forward = new Forward();
        this.midfielder = new Midfielder();
    }

    executeGamePlan() {
        this.forward.attack();
        this.midfielder.controlMidfield();
    }
}
module.exports = Coach;

File: Chapter 03/C#/3.3.5 DIP/Before/Coach.cs
namespace Chapter03.DIP.Before;

public class Coach
{
    // The Coach is tightly coupled to concrete classes!
    private Forward _forward;
    private Midfielder _midfielder;

    public Coach()
    {
        _forward = new Forward();
        _midfielder = new Midfielder();
    }

    public void ExecuteGamePlan()
    {
        _forward.Attack();
        _midfielder.ControlMidfield();
    }
}